lib/utils/scope-subtree.js

```diff
@@ -0,0 +1,76 @@
+/**
+@license
+Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+import './boot.js';
+import { wrap } from './wrap.js';
+
+const ShadyDOM = window.ShadyDOM;
+const ShadyCSS = window.ShadyCSS;
+
+/**
+ * Ensure that elements in a ShadowDOM container are scoped correctly.
```

Comment:
Let's add a note about when this is needed: only when ShadyDOM is used and unpatched DOM APIs are used in third party code. This can happen in noPatch mode or when specialized APIs like ranges or tables are used to mutate DOM.

Reply:
Done.